rails_4_post_mortem.html

```diff
@@ -0,0 +1,109 @@
+<!doctype html>
+<html lang="en">
+
+  <head>
+    <meta charset="utf-8">
+
+    <title>Rails 4 Post Mortem</title>
+
+    <meta name="apple-mobile-web-app-capable" content="yes" />
+    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
+
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
+
+    <link rel="stylesheet" href="lib/reveal.js/css/reveal.min.css">
+    <link rel="stylesheet" href="lib/reveal.js/css/theme/night.css">
+    <link rel="stylesheet" href="lib/template.css">
+
+    <!-- For syntax highlighting -->
+    <link rel="stylesheet" href="lib/reveal.js/lib/css/zenburn.css">
+
+    <!-- If the query includes 'print-pdf', include the PDF print sheet -->
+    <script>
+      if( window.location.search.match( /print-pdf/gi ) ) {
+        var link = document.createElement( 'link' );
+        link.rel = 'stylesheet';
+        link.type = 'text/css';
+        link.href = 'css/print/pdf.css';
+        document.getElementsByTagName( 'head' )[0].appendChild( link );
+      }
+    </script>
+
+    <!--[if lt IE 9]>
+    <script src="lib/reveal.js/lib/js/html5shiv.js"></script>
+    <![endif]-->
+  </head>
+
+  <body>
+    <div class="reveal">
+      <div class="slides" id="content">
+        <section data-markdown  
+                 data-separator="==="
+                 data-vertical="---"><script type="text/template">
+
+# Rails 4 Post Mortem
+
+What can we learn from the upgrade to Rails 4
+
+===
+
+## Tests
+
+---
+
+### Some critical features aren't tested
+
+* More routing tests (verbs, permalinks, parameters, SP*, I18n)
+* More API tests (XML)
+* Critical paths (Account activation, password reset): The current scenario have been tested by hand and are not covering enough
+
+---
+
+### More human tests?
+
+* The dedicated server has not been used
+* Who should make live tests?
+
+===
+
+## Maintenance
+
+* More people **(live!)**
+* Non regression **(live!)**
+
+
+
+---
+
+### Where do bugs come from
+
+* Check what is the weakest layer of our application (models, controllers, views)
+* Improve test coverage for that layer in priority
+
+---
+
+### Maintainability
+
+* Some bugs were fixed multiple times because changing one part of the code made an other thing break *(ex: Elasticsearch and his infamous callback hell)*
```

Comment:
Don't really agree on the "Elasticsearch and his infamous callback hell"
Callbacks are generated by [Tire](https://github.com/karmi/retire/blob/master/lib/tire/model/callbacks.rb#L20), which is not the best wrapper for the ES api. So, yes Tire is a bottleneck, but maybe not ES.